pageserver/src/tenant/timeline.rs

```diff
@@ -911,6 +911,43 @@ impl Timeline {
         }
     }
 
+    fn validate_key_equivalence(
+        key: &Key,
+        keyspace: &KeySpace,
+        lsn: Lsn,
+        seq: &Bytes,
+        vec: &Bytes,
+    ) {
+        use utils::bin_ser::BeSer;
+
+        if *key == AUX_FILES_KEY {
+            // The value reconstruct of AUX_FILES_KEY from records is not deterministic
+            // since it uses a hash map under the hood. Hence, deserialise both results
+            // before comparing.
+            let seq_aux_dir_res = AuxFilesDirectory::des(seq);
+            let vec_aux_dir_res = AuxFilesDirectory::des(vec);
+            match (&seq_aux_dir_res, &vec_aux_dir_res) {
+                (Ok(seq_aux_dir), Ok(vec_aux_dir)) => {
+                    assert_eq!(
+                        seq_aux_dir, vec_aux_dir,
+                        "Mismatch for key {} - keyspace={:?} lsn={}: {:?} != {:?}",
+                        key, keyspace, lsn, seq_aux_dir, vec_aux_dir
+                    );
```

Comment:
```suggestion
                    assert_eq!(
                        seq_aux_dir, vec_aux_dir,
                        "Mismatch for key {} - keyspace={:?} lsn={}",
                        key, keyspace, lsn, seq_aux_dir, vec_aux_dir
                    );
```

The values are already debug formatted by the macro (and I guess on any problem they'd be large): https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3ccf6c23a34347f1b684e9386610eb09

Or did you find there's some conditionality? I would guess no such thing.

Reply:
The implementation of `assert_eq!` looks like it does not react to size of left and right.

For the other assert, the `{:?}` of Bytes produces good messages.

Reply:
Done in https://github.com/neondatabase/neon/pull/7018/commits/a9e0d09e80b01e86267510bae6b98faaf137df6b